src/include/migraphx/op/roialign.hpp

```diff
@@ -0,0 +1,321 @@
+            const auto* batch_indices = args.at(2).cast<int64_t>();
+            par_for(n_rois, [&](auto n) {
+                const auto* bottom_data  = x.data();
+                const auto roi_batch_ind = batch_indices[n];
+                // Do not using rounding; this implementation detail is critical
+                float roi_start_w = static_cast<float>(roi[roi_s.index({n, 0})] * spatial_scale);
+                float roi_start_h = static_cast<float>(roi[roi_s.index({n, 1})] * spatial_scale);
+                float roi_end_w   = static_cast<float>(roi[roi_s.index({n, 2})] * spatial_scale);
+                float roi_end_h   = static_cast<float>(roi[roi_s.index({n, 3})] * spatial_scale);
+
+                // Force malformed ROIs to be 1x1
+                float roi_width =
+                    (roi_end_w - roi_start_w) > 1.0f ? (roi_end_w - roi_start_w) : 1.0f;
+                float roi_height =
+                    (roi_end_h - roi_start_h) > 1.0f ? (roi_end_h - roi_start_h) : 1.0f;
+                float bin_size_h = static_cast<float>(roi_height / pooled_height);
+                float bin_size_w = static_cast<float>(roi_width / pooled_width);
+
+                // We use roi_bin_grid to sample the grid and mimic integral
+                int64_t roi_bin_grid_h =
+                    (sampling_ratio > 0)
+                        ? sampling_ratio
+                        : static_cast<int64_t>(std::ceil(roi_height / pooled_height));
+                int64_t roi_bin_grid_w =
+                    (sampling_ratio > 0)
+                        ? sampling_ratio
+                        : static_cast<int64_t>(std::ceil(roi_width / pooled_width));
+
+                // we want to precalculate indices and weights shared by all channels,
+                // this is the key point of optimization
+                std::vector<pos_weight> pre_calc(roi_bin_grid_h * roi_bin_grid_w * pooled_width *
+                                                 pooled_height);
+                std::vector<int64_t> lens = {
+                    pooled_height, pooled_width, roi_bin_grid_h, roi_bin_grid_w};
+                std::vector<std::size_t> comp_lens(lens.begin(), lens.end());
+                shape comp_s{shape::float_type, comp_lens};
+                this->calc_pos_weight(height,
+                                      width,
+                                      comp_s,
+                                      roi_start_h,
+                                      roi_start_w,
+                                      bin_size_h,
+                                      bin_size_w,
+                                      roi_bin_grid_h,
+                                      roi_bin_grid_w,
+                                      pre_calc);
+
+                std::vector<int64_t> lens1 = {channels, pooled_height, pooled_width};
+                std::vector<std::size_t> comp_lens1(lens1.begin(), lens1.end());
```

Comment:
Wouldn't it implicitly cast if you say 
`vector<size_t> comp_lens1 = {channels, pooled_height, pooled_width}` ?

Reply:
I remember there is an build error for this case when I did the rewrite_rnn, see:
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/8829d6ab2b75dd5c4e223029b83457f59de71e5b/src/rewrite_rnn.cpp#L258

Looks like it works fine now, just fixed it.